src/main/java/net/minecraftforge/client/ItemOverlayHandler.java

```diff
@@ -0,0 +1,123 @@
+package net.minecraftforge.client;
+
+import java.util.List;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.VertexBuffer;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.texture.TextureManager;
+import net.minecraft.client.renderer.texture.TextureMap;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.common.ForgeModContainer;
+
+/**
+ * Offshore handler for the Items' extra passes. Specifically for reducing patch size & maintaining readability.
+ * */
+public final class ItemOverlayHandler
+{
+
+    private static final java.util.Map<net.minecraftforge.fml.common.registry.RegistryDelegate<Item>, IItemOverlay> itemOverlayMap = com.google.common.collect.Maps.newHashMap();
+
+    private static final byte NUM_PASS = 2;
+
+    private ItemOverlayHandler(){}
+
+    /**
+     * Handles the execution of overlays.
+     * 
+     * @param textureManager the current TextureManager
+     * @param stack the stack in question
+     * @param model 
+     * 
+     * @return true to render the item again, and false to end the pass loop.
+     * */
+    public static boolean run(TextureManager textureManager, ItemStack stack, IBakedModel model)
+    {
+        if(ForgeModContainer.forgeItemOverlay)
+        {
+            float[] vec;
+            int color;
+            float time;
+            IItemOverlay current = itemOverlayMap.get(stack.getItem().delegate);
+            if(current == null)
+               current = IItemOverlay.VANILLA;
+            GlStateManager.depthMask(false);
+            GlStateManager.depthFunc(514);
+            GlStateManager.disableLighting();
+            GlStateManager.matrixMode(5890);
+            textureManager.bindTexture(current.getOverlayTexture(stack));
+            
+            for(int CUR_PASS = 0; CUR_PASS < NUM_PASS; CUR_PASS++)
+            {
+                color = current.getOverlayColor(stack, CUR_PASS);
+                time = current.getShiftedTime(stack, CUR_PASS, Minecraft.getSystemTime());
+                if(((color >> 16 & 255) | (color >> 8 & 255) | (color & 255)) >= 0xA1)
```

Comment:
Because the default blending caused dark values to manifest as transparency. When values were below the threshold they'd be barely visible. I don't know about the exact math of the matter but it's more or less fixed. Still doesn't support true black but that's due to the texture's never having values that reflect full opacity. May have to test that now :P